### PR TITLE
some dashboard refactor

### PR DESCRIPTION
**Pull request by hbrwang**
* upgrade vue-cli 4
* add eslint
* refactor requests

### Compiles and hot-reloads for development
```
npm run serve // not dev in vue-cli4
```

> 1. 目前vue-cli4是主流，vue-cli4环境无法运行vue-cli2项目。升级项目以便他人运行。
> 2. 使用 eslint 规范代码。

### CODE REVIEW

**Comment by fatedier** [comment]
Can we just run `make build` to generate files?

We'd better auto install thrid party dependencies. Write the steps into `build.sh` if it's complicated.

**Comment by fatedier** [comment]
A simple question: We haven't used vue-cli before, why we should import it? What's the benefits?

**Comment by hbrwang** [comment]
> A simple question: We haven't used vue-cli before, why we should import it? What's the benefits?

vue-cli is more simpler and efficient. Using vue-cli is a more elegant way.

---

> Can we just run `make build` to generate files?
>
> We'd better auto install thrid party dependencies. Write the steps into `build.sh` if it's complicated.

I have't use `golang` and `make` before, but the `Makefile` is edited. I'm trying to do that now.

**Comment by fatedier** [comment]
OK. Ping me later if you updated.

**Comment by hbrwang** [comment]
Thank you for giving me the opportunity to learn `makefile`. It's amazing!

I ran `bash package.sh`,  and built files without error.

Then I tested the `windows_amd64` and `linux_amd64`. It seems all right.

_I didn't edit any files_

File: web/frps/Makefile [hunk 1]
```diff
@@ -4,4 +4,4 @@ build:
 	@npm run build
```
Comment (fatedier) [inline web/frps/Makefile:4]: We should provide build scripts here to let others build it in a simple way.

I run `make build` on my machine and it failed. If it relies on other tools, we'd better descirbe it on README.

Comment (hbrwang) [reply web/frps/Makefile:4]: I just edited the `Makefile` to  `install` before `build`

Comment (fatedier) [reply web/frps/Makefile:4]: `yarn install` is used in README.md, what's the difference?

Comment (hbrwang) [reply web/frps/Makefile:4]: It was automatically generated and has now been edited

**Review by fatedier — APPROVED** [review]

**Comment by fatedier** [comment]
@hbrwang Can you do the same refactor for frpc web UI?

**Comment by hbrwang** [comment]
Sure, it's my pleasure

**Comment by fatedier** [comment]
@hal-wang From this PR, the binary file size is much more greater than previous release.

I add `productionSourceMap: false` in `vue.config.js` to remove map files, but it still greater than before.

The web assets's size is 2MB now, before is 1.3MB.

I'm not sure if it's releated to vue-cli 4. If so, i will revert it to original state.

**Comment by hal-wang** [comment]
It seems that vue-cli is larger than webpack, and a better solution is to use SDN. The PR #2129 ,assets's size is 0.6MB

**Comment by fatedier** [comment]
@hal-wang It should not rely on any external resources. We compile all assets into binary files, so users can visit dashboard in any environment even without public network.

I will revert web assets to original state first. We can refactor later without making the web assets larger.

**Comment by hal-wang** [comment]
@fatedier  OK, sorry for the trouble

**Comment by fatedier** [comment]
@hal-wang I'm not familiar with front code. I'm confused why file size is grater than before that we add no new featrues.

Can we change to vue-cli without file size changed? If we ignored some configures?

**Comment by hal-wang** [comment]
@fatedier There should be a problem with the configuration. Let me have a further study.

**Pull request merged**